### ISSUE TITLE
[Feature Request] Include type hints in function parameters?

### ISSUE THREAD

**Issue by ldorigo** [plotly/dash#2233]
Hi, 

Static type checkers (mypy, pyright/pylance) are becoming increasingly popular in the python world.
For now they don't work with Dash as there are no type annotations included with the library.

From what I understand, python code for dash components as well as their docstrings are somehow transpiled from javascript? If so, adding type hints should be relatively straightforward as it appears that the docstrings already contain very detailed typing information. This would significantly improve QOL for users using type checkers (which includes the majority of the vscode userbase since vscode uses pylance by default).

For reference, plotly has also started adding typing information to their functions - see for instance https://github.com/plotly/plotly.py/pull/3425#issuecomment-1117210067 .

**Comment by tuchandra** [comment]
Huge +1 from me; the developer experience of using a library that type hints its parameters is unmatched. There's some related work I've stumbled on:
- #1748 was a draft PR to add type annotations generally
- #1956 (merged!) supports generating components from TypeScript, which in theory should be even more powerful than what you describe
- #2170 is about a specific type hinting request

**Comment by ajrheaume** [comment]
Type hints would be super useful! Any update about this feature request?

**Comment by gvwilson** [comment]
@T4rk1n will your work on the code generator take care of this as a side effect?

**Comment by T4rk1n** [comment]
> @T4rk1n will your work on the code generator take care of this as a side effect?

No, it is a task on it's own to add the typing to all API methods and then there is #2170 which are all typing related.
